tardis/transport/montecarlo/formal_integral_cuda.py

```diff
@@ -12,6 +12,212 @@
 H_CGS = 6.62606957e-27
 
 
+@cuda.jit
+def cuda_vector_integrator(L, I_nu, N, R_max):
+
+    nu_idx = cuda.grid(1)
+    L[nu_idx] = (
+        8 * M_PI * M_PI * trapezoid_integration_cuda(I_nu[nu_idx], R_max / N)
+    )
+
+@cuda.jit
+def cuda_formal_integral_fast(
+    r_inner,
+    r_outer,
+    time_explosion,
+    line_list_nu,
+    iT,
+    inu,
+    inu_size,
+    att_S_ul,
+    Jred_lu,
+    Jblue_lu,
+    tau_sobolev,
+    electron_density,
+    N,
+    pp,
+    exp_tau,
+    I_nu,
+    z,
+    shell_id,
+):
+    """
+    The CUDA version of numba_formal_integral that can run
+    on a NVIDIA GPU.
+
+    Parameters
+    ----------
+    r_inner : array(float64, 1d, C)
+        self.geometry.r_inner
+    r_outer : array(float64, 1d, C)
+        self.geometry.r_outer
+    time_explosion: float64
+        self.geometry.time_explosion
+    line_list_nu : array(float64, 1d, A)
+        self.plasma.line_list_nu
+    iT : np.float64
+    inu : np.float64
+    inu_size : int64
+    att_S_ul : array(float64, 1d, C)
+    Jred_lu : array(float64, 1d, C)
+    Jblue_lu : array(float64, 1d, C)
+    tau_sobolev : array(float64, 2d, C)
+    electron_density : array(float64, 1d, C)
+    N : int64
+    L : array(float64, 1d, C)
+        This is where the results will be stored
+    pp : array(float64, 1d, C)
+    exp_tau : array(float64, 1d, C)
+    I_nu array(floatt64, 2d, C)
+    z : array(float64, 2d, C)
+    shell_id : array(int64, 2d, C)
+    """
+
+    # todo: add all the original todos
```

Comment:
Probably out of date now...

Reply:
Yeah, I have no idea even where they are anymore.  I'll remove it